server/controllers/profile.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const Profile = require('../models/Profile');
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile
+// @desc Create profile
+// @access Public
+exports.createProfile = asyncHandler(async (req, res, next) => {
+    // Checking for empty input
+    for (let key in req.body) {
+        if (req.body[key].trim() === "") {
+          res.status(406);
+          throw new Error("Invalid input, please do not send empty input(s)")
+        }
+      }
+
+    const profile = new Profile(req.body);
+
+    try {
+        await profile.save(); 
+        res.status(201).json({ success: { profile }});
+    } catch(error) {
+        res.status(400).json({ error });
+    }
```

Comment:
Good job on using `save` as it is the preferred way to update documents 🙌🏼 

This `try...catch` block is how we normally implement code blocks with `await`. But in this case, it's already done for us thanks to the `express-async-handler` package (i.e. on `line 8` we use `asyncHandler()` method). It basically catches any errors thrown and forwards it to the next handler (which is the `error` middleware in our case).